api/server/sdk/volume_snapshot.go

```diff
@@ -0,0 +1,98 @@
+/*
+Package sdk is the gRPC implementation of the SDK gRPC server
+Copyright 2018 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package sdk
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/libopenstorage/openstorage/api"
+)
+
+// SnapshotCreate creates a read-only snapshot of a volume
+func (s *VolumeServer) SnapshotCreate(
+	ctx context.Context,
+	req *api.VolumeSnapshotCreateRequest,
+) (*api.VolumeSnapshotCreateResponse, error) {
+
```

Comment:
ignoring input ctx? i would recommend checking if input ctx deadline has been requested by the client and take any appropriate action.

Reply:
Yeah, we are going to do that for all grpc calls (CSI ones also) at a future change.